kinoml/workflows/chembl22_cnn.py

```diff
@@ -0,0 +1,188 @@
+        else:
+            print(f'Shape of input: {x.shape} and output: {y.shape} .')
+
+
+        # Metric per fold (K values)
+        acc_per_fold = []
+        conf_matrix_per_fold = []
+
+        # K fold Cross Validation
+        kfold = KFold(n_splits=K, random_state=seed_cv)
+        for j, (train_index, test_index) in enumerate(kfold.split(x, y)):
+
+            print(f'Fold {j}')
+
+            x_train, x_test = x[train_index], x[test_index]
+            y_train, y_test = y[train_index], y[test_index]
+
+            print('---Train')
+
+            for epoch in range(nb_epoch):
+                
+                # Clear the previous gradients
+                optimizer.zero_grad()
+
+                # Feedforward pass
+                outputs_train = torch.squeeze(model(x_train)) # Use torch.squeeze to remove axis with 1 dimension
+                if outputs_train.shape != y_train.shape:
+                    print('The shape of the true value of the target differs..'
+                          '..from the shape of the value predicted by the model.')
+                if torch.min(outputs_train)<0 or torch.max(outputs_train)>1:
+                    print('The predicted values are out of range.')
+                    
+                if all(outputs_train[0] == elem for elem in outputs_train):
+                    print(f'The model predicts all values at {outputs_train[0]} .')
+
+                loss = criterion(outputs_train, y_train)
+                loss.backward()
+                optimizer.step()
+
+            print('---Predict')
+            # Test trained model on the test set
+            outputs_test = torch.squeeze(model(x_test))
+
+            # Transform target vectors into numpy array for scores
+            y_pred = outputs_test.detach().numpy()
+            y_test = y_test.detach().numpy()
+
+            # Transform probability of belonging to one class into 0 or 1 value at threshold 0.5
+            y_pred =(y_pred>0.5) 
+
+            if(all(y_pred[0] == elem for elem in y_pred)):
+                print(f'The model predicts all values at {y_pred[0]}')
+
+            print(f'# of elements on the test set: {len(y_test)}, # zero counts on the test set: {(y_test==0).sum()} ')
+
+            # Metrics on each fold
+            acc_per_fold.append(accuracy_score(y_test, y_pred))
+            conf_matrix_per_fold.append(confusion_matrix(y_test, y_pred).ravel())
+
+        # Mean on all K folds
+        acc_mean = np.mean(acc_per_fold)
+        conf_matrix_mean = np.mean(np.array(conf_matrix_per_fold), axis=0)
+
+        # Std on all K folds
+        acc_std = np.std(acc_per_fold)
+        conf_matrix_std = np.std(np.array(conf_matrix_per_fold), axis=0)
+
+        print(f' Metrics of CV for kinase {kinase} ')
+        print(f' Accuracy : Mean : {round(acc_mean, 2)} and std : {round(acc_std, 2)} ')
+        print(f' TN : Mean : {round(conf_matrix_mean[0], 2)} and std : {round(conf_matrix_std[0], 2)} ')
```

Comment:
You can use format notation for floats to easily get the rounded quantity. For example, `f'{acc_mean:.2f}` means "print `acc_mean` as a float with two decimal positions. Check the _new_ style in [this website](https://pyformat.info/).